kornia/filters/kernels.py

```diff
@@ -14,14 +16,20 @@ def normalize_kernel2d(input: torch.Tensor) -> torch.Tensor:
 
 
 def gaussian(window_size: int, sigma: float) -> torch.Tensor:
-    device, dtype = None, None
-    if isinstance(sigma, torch.Tensor):
-        device, dtype = sigma.device, sigma.dtype
-    x = torch.arange(window_size, device=device, dtype=dtype) - window_size // 2
+    sigma_t = torch.tensor(sigma)
+
+    return gaussian_t(window_size, sigma_t)
+
+
+def gaussian_t(window_size: int, sigma: torch.Tensor) -> torch.Tensor:
```

Comment:
```suggestion
def gaussian_t(window_size: int, sigma: Tensor) -> Tensor:
```
`from kornia.core import Tensor`